### ISSUE TITLE
learnr is loaded on startup if installed

### ISSUE THREAD

**Issue by kevinushey** [rstudio/rstudio#7265]
### System details

    RStudio Edition : Desktop
    RStudio Version : v1.3.1053, v1.4.x
    OS Version      : All
    R Version       : All

### Steps to reproduce the problem

Install the `learnr` package, e.g. via:

```
install.packages("learnr")
```

Then, restart the R session, and look at what packages are loaded via `loadedNamespaces()`.

### Describe the problem in detail

`learnr` and its dependencies are loaded on startup:

```
> loadedNamespaces()
 [1] "compiler"  "graphics"  "htmltools" "tools"     "parallel"  "utils"     "yaml"      "grDevices"
 [9] "stats"     "datasets"  "rmarkdown" "knitr"     "methods"   "xfun"      "digest"    "packrat"  
[17] "rlang"     "base"      "evaluate" 
```

### Describe the behavior you expected

We should avoid aggressively loading packages on startup when possible; especially so on Windows where updates of installed packages can fail.

### Notes

We can put the following hook in `~/.Rprofile` to get a traceback from when `rmarkdown` is loaded:

```
  setHook(
    packageEvent("rmarkdown", "onLoad"),
    function(...) print(rlang::trace_back())
  )
```

This shows:

```
 █
  1. └─(function (pkgRoot) ...
  2.   └─base::lapply(tutorialDirs, .rs.tutorial.findTutorialsImpl)
  3.     └─FUN(X[[i]], ...)
  4.       └─rmarkdown:::parse_yaml_front_matter
  5.         ├─base::get(name, envir = asNamespace(pkg), inherits = FALSE)
  6.         └─base::asNamespace(pkg)
  7.           └─base::getNamespace(ns)
  8.             └─base::loadNamespace(name)
  9.               └─base:::runUserHook(package, pkgpath)
 10.                 ├─base::try(fun(pkgname, pkgpath))
 11.                 │ └─base::tryCatch(...)
 12.                 │   └─base:::tryCatchList(expr, classes, parentenv, handlers)
 13.                 │     └─base:::tryCatchOne(expr, names, parentenv, handlers[[1L]])
 14.                 │       └─base:::doTryCatch(return(expr), name, parentenv, handler)
 15.                 └─fun(pkgname, pkgpath)
```

And so we can ascertain that `rmarkdown` is being loaded and used while attempting to enumerate available tutorials. We should see if we can avoid needing to load `rmarkdown` to enumerate tutorials here.

---

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
Note: the call to `rmarkdown:::parse_yaml_front_matter()` is here:

https://github.com/rstudio/rstudio/blob/b65bc9fbe62c8fc2ae9e984006de4094442bb1b1/src/cpp/session/modules/SessionTutorial.R#L250

and we could probably use our internal YAML parser / tools to find and read the YAML here.

**Comment by R-Hannibal** [comment]
Verified Fixed on: RStudio Version - 1.3.1060, R Version - 4.0.2, Windows 10 x64 - Desktop

Comparison tested versus 1.3.1056, listing the output below but it looks like `learnr` and its dependencies are no longer being loaded on session startup:
```
> loadedNamespaces()
 [1] "compiler"  "graphics"  "tools"     "utils"     "yaml"      "grDevices"
 [7] "stats"     "datasets"  "methods"   "base"
```
